src/plugins/posix/posix_backend.cpp

```diff
@@ -0,0 +1,311 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <liburing.h>
+#include <algorithm>
+#include <cmath>
+#include "posix_backend.h"
+#include <unordered_map>
+#include <absl/log/log.h>
+
+namespace {
+    static constexpr int max_posix_ring_log = 6; //posix_ring_log
+    static constexpr int max_posix_ring_size = 1 << max_posix_ring_log;
+    const nixl_mem_list_t supported_mems = {
+        FILE_SEG,
+        DRAM_SEG
+    };
+}
+
+void nixlPosixBackendReqH::createUringParams() {
+    // We only need 2 params, as only the last one will be the remainder, and others will be max_posix_ring_size

```

Comment:
What is the max_posix_ring_size? and how many rings are we creating.
If we submit 12K requests how many rings does that map to?